s3transfer/download.py

```diff
@@ -79,6 +83,27 @@ def get_fileobj_for_io_writes(self, transfer_future):
         """
         raise NotImplementedError('must implement get_fileobj_for_io_writes()')
 
+    def queue_file_io_task(self, fileobj, data, offset):
+        """Queue IO write for submission to the IO executor.
+
+        This method accepts an IO executor and information about the
+        downloaded data, and handles submitting this to the IO executor.
+
+        This method may defer submission to the IO executor if necessary.
+
+        """
```

Comment:
I agree with this approach as we really would not want to pass in a reference to the ``GetObjectTask`` to invoke the ``_submit_task``. But should we put in a log statement as well to achieve better parity with the ``_submit_task``: https://github.com/boto/s3transfer/blob/develop/s3transfer/tasks.py#L200?